lib/smart_answer/calculators/state_pension_age_calculator.rb

```diff
@@ -16,6 +16,19 @@ def state_pension_date
       StatePensionDateQuery.state_pension_date(dob, gender)
     end
 
+    def approaching_pension_age?
```

Comment:
@floehopper I understand your context here.. the naming was influence from my discovery session with @raykhan66.. I feel it better covers the `within` context of this scenario.